src/io/ioctl.rs

```diff
@@ -42,7 +45,12 @@ pub fn ioctl_fioclex<Fd: AsFd>(fd: Fd) -> io::Result<()> {
 #[inline]
 #[doc(alias = "FIONBIO")]
 pub fn ioctl_fionbio<Fd: AsFd>(fd: Fd, value: bool) -> io::Result<()> {
-    backend::io::syscalls::ioctl_fionbio(fd.as_fd(), value)
+    // SAFETY: FIONBIO is a pointer setter opcode.
+    #[allow(unsafe_code)]
+    let ctl = unsafe {
+        ioctl::PtrSetter::<ioctl::BadOpcode<{ c::FIONBIO }>, c::c_int>::new(c::c_int::from(value))
```

Comment:
Instead of `PtrSetter`, could it be called just `Setter`? From the perspective of code defining an ioctl like this, the fact that it uses a pointer internally isn't what's most important.

Why is this using a "bad" opcode? `FIONBIO` seems like a perfectly good opcode to me ;-).

Instead of naming `c::c_int` twice, could `c::c_int::from(value)` be `value.into()`?

Reply:
> Why is this using a "bad" opcode? FIONBIO seems like a perfectly good opcode to me ;-).

"Bad" opcode refers to the "old" strategy of handling opcodes, where it's just a number instead of a composition of the IOCTL code, subtype, direction, etc etc.

Reply:
Could we call them "old" opcodes instead? And add a comment explaining that "old" refers to how the opcode is encoded as just a number?